src/sentry/relay/config/metric_extraction.py

```diff
@@ -291,10 +306,40 @@ def _trim_if_above_limit(
                 )
 
             return_specs += specs_for_version[:max_specs]
+            if (
+                OnDemandMetricSpecVersioning.get_query_spec_version(project.organization.id)
+                == SpecVersion(1)
```

Comment:
What's the purpose of this check?

You may want to do `OnDemandMetricSpecVersioning.spec_versions[0]` or `OnDemandMetricSpecVersioning.spec_versions[-1]` depending if you want the current version or the latest version.

Reply:
We don't need it now that there are multiple spec rows.